app/models/competition_event.rb

```diff
@@ -32,7 +32,7 @@ def currency_code
     competition&.currency_code
   end
 
-  def has_fee?
+  def payment_required?
```

Comment:
Same reasoning as above: `payment_required?` is much too broad.
Whatever you come up with on the bigger `competition` object can probably be used here as well.